scripts/sync-ros2-gbp-devel-branch.py

```diff
@@ -0,0 +1,219 @@
+        if not release_url.startswith('https://github.com'):
+            print("Release URL {release_url} for package '{repo}' is not on GitHub, do not know how to fetch tracks.yaml data".format(release_url=release_url, repo=repo))
+            continue
+
+        constrained_list.append(repo_dict)
+
+    return constrained_list
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--all-repos', help='Check all repositories, not just core ROS 2 ones', action='store_true', default=False)
+    parser.add_argument('--dry-run', help='Just print the differences, do not actually open PRs', action='store_true', default=False)
+    parser.add_argument('distribution', nargs=1, help='Which ROS distribution to do the sync for', action='store')
+    args = parser.parse_args()
+
+    key = keyring.get_password('github-open-prs', 'may-open-prs')
+    if key is None:
+        raise RuntimeError('Failed to get GitHub API key')
+
+    gh = github.Github(key)
+
+    ros_distro = args.distribution[0]
+
+    # First get the rosdistro distribution.yaml, which we will use as the source
+    # of the devel_branch we should use.
+    rosdistro_url = 'https://raw.githubusercontent.com/ros/rosdistro/master/{ros_distro}/distribution.yaml'.format(ros_distro=ros_distro)
+
+    with urllib.request.urlopen(rosdistro_url) as response:
+        ros_distro_data = response.read()
+
+    ros_distro_yaml = yaml.safe_load(ros_distro_data)
+
+    if args.all_repos:
+        constrained_list = get_all_ros2_repositories(ros_distro_yaml)
+    else:
+        constrained_list = get_ros2_core_repositories(ros_distro, ros_distro_yaml)
+
+    # Now that we have the list of repositories constrained, iterate over each
+    # one, comparing what is in the tracks.yaml in the release repository to
+    # what is in the source entry in the <distro>/distribution.yaml
+    for repo in constrained_list:
+        release_url = repo['release']['url']
+
+        release_end = release_url[19:-4]
+        tracks_url = 'https://raw.githubusercontent.com/' + release_end + '/master/tracks.yaml'
+
+        with urllib.request.urlopen(tracks_url) as response:
+            tracks_data = response.read()
+
+        tracks_yaml = yaml.safe_load(tracks_data)
+        tracks_yaml_distro = tracks_yaml['tracks'][ros_distro]
+
+        if tracks_yaml_distro['devel_branch'] != repo['source']['version']:
+            print("Package '{reponame}' rosdistro source branch ({source_branch}) does not match release branch ({release_branch})".format(reponame=tracks_yaml_distro['name'], source_branch=repo['source']['version'], release_branch=tracks_yaml_distro['devel_branch']))
+
+            if args.dry_run:
+                continue
+
+            gh_body = """This PR from an automated script updates the devel_branch for {ros_distro} to match the source branch as specified in https://github.com/ros/rosdistro/{ros_distro}/distribution.yaml .
+
+Links to https://github.com/ros2/ros2/issues/963 .
```

Comment:
Does it make sense to keep this issue link if the issue is closed?

Reply:
Yeah, that's a good question.  I had a vague idea that if we were ever to use this script again, we could just reopen that issue.

Maybe a better idea is to have a command-line argument that allows the user to pass the link they want to use in.  What do you think?

Reply:
I'm generally on team don't add options you don't already need. I think it'd make more sense to first PR the script to use a different issue body.

Reply:
Should be fixed in 2db8383